diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt

```diff
@@ -231,10 +231,12 @@ class NullChecksRule(configRules: List<RulesConfig>) : DiktatRule(
     /**
      * checks if condition is a complex expression. For example:
      * (a == 5) - is not a complex condition, but (a == 5 && b != 6) is a complex condition
+     * or expression could be used in lambda, which is also considered as complex expression:
+     * if (a.any { it == null })
      */
     private fun KtBinaryExpression.isComplexCondition(): Boolean {
-        // KtBinaryExpression is complex if it has a parent that is also a binary expression
-        return this.parent is KtBinaryExpression
+        // KtBinaryExpression is complex if it has a parent that is a binary expression or a block expression
+        return this.parent is KtBinaryExpression || this.parent is KtBlockExpression
```

Comment:
I think we should rather check only condition instead of `findAllDescendantsWithSpecificType` in `conditionInIfStatement`, than change logic here. We shouldn't have gotten to this line in the first place, because comparison with `null` inside lambda is not a part `if` condition, that our rule is trying to change.

Reply:
Moved into separate method and added test as discussed